src/compiler/checker.ts

```diff
@@ -15638,8 +15651,14 @@ namespace ts {
                 return getStringLiteralType(text);
             }
             newTexts.push(text);
-            if (every(newTexts, t => t === "") && every(newTypes, t => !!(t.flags & TypeFlags.String))) {
-                return stringType;
+            if (every(newTexts, t => t === "")) {
+                if (every(newTypes, t => !!(t.flags & TypeFlags.String))) {
+                    return stringType;
+                }
+                // Normalize `${Mapping<xxx>}` into Mapping<xxx>
+                if (newTypes.length === 1 && isPatternLiteralType(newTypes[0])) {
+                    return newTypes[0];
+                }

```

Comment:
Is this correct for if `xxx` becomes `number`?

Reply:
The string mapping types are all constrained to `string`, so `xxx` can't be `number`. It could however be `` `${number}` ``.